hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElementType.java

```diff
@@ -130,18 +129,20 @@ public Queryable getQueryable() {
 	String renderScalarIdentifierSelect(int i) {
 		checkInitialized();
 
-		final String idPropertyName = getIdentifierPropertyName();
-		String[] cols = getPropertyMapping( idPropertyName ).toColumns( getTableAlias(), idPropertyName );
-
+		final String[] idPropertyName = getIdentifierPropertyNames();
 		StringBuilder buf = new StringBuilder();
-		// For property references generate <tablealias>.<columnname> as <projectionalias>
-		for ( int j = 0; j < cols.length; j++ ) {
-			String column = cols[j];
-			if ( j > 0 ) {
-				buf.append( ", " );
+		for ( int j = 0; j < idPropertyName.length; j++ ) {
+			String propertyName = idPropertyName[j];
+			String[] toColumns = getPropertyMapping( propertyName ).toColumns( getTableAlias(), propertyName );
+			for ( int h = 0; h < toColumns.length; h++ ) {
+				String column = toColumns[h];
+				if ( j + h > 0 ) {
+					buf.append( ", " );
+				}
+				buf.append( column ).append( " as " ).append( NameGenerator.scalarName( i, j + h ) );
```

Comment:
Using `j+h` above can result in duplicated column aliases. For example, if `i == 0`, `idPropertyName.length == 2` and each `toColumns.length == 2`, then the column names that will be generated with be: `col_0_0_`, `col_0_1_`, `col_0_1`, `col_0_2`.

The column names should be `col_0_0_`, `col_0_1_`, col_0_2_`, `col_0_3_`.

I've created a new PR with a failing test case that illustrates this, and added a commit for the fix: 
https://github.com/hibernate/hibernate-orm/pull/2685